src/Build.UnitTests/BuildEnvironmentHelper_Tests.cs

```diff
@@ -59,6 +59,41 @@ public void FindBuildEnvironmentByEnvironmentVariable()
             }
         }
 
+#if RUNTIME_TYPE_NETCORE
+        [Theory(Skip = "NA on .net core")]
+#else
+        [Theory]
+#endif
+        [InlineData(true)]
+        [InlineData(false)]
+        [Trait("Category", "nonlinuxtests")]
+        [Trait("Category", "nonosxtests")]
+        public void FindVisualStudioEnvironmentByEnvironmentVariable(bool is64BitMSbuild)
+        {
+            using (var env = new EmptyVSEnviroment())
+            {
+                var msbuildBinDirectory = is64BitMSbuild
+                    ? Path.Combine(env.BuildDirectory, "amd64")
+                    : env.BuildDirectory;
+
+                var msBuildPath = Path.Combine(msbuildBinDirectory, MSBuildExeName);
+                var msBuildConfig = Path.Combine(msbuildBinDirectory, $"{MSBuildExeName}.config");
+                var vsMSBuildDirectory = Path.Combine(env.TempFolderRoot, "MSBuild");
+
+                env.WithEnvironment("MSBUILD_EXE_PATH", msBuildPath);
+                BuildEnvironmentHelper.ResetInstance_ForUnitTestsOnly(ReturnNull, ReturnNull, ReturnNull, env.VsInstanceMock, env.EnvironmentMock, () => false);
+
+                BuildEnvironmentHelper.Instance.Mode.ShouldBe(BuildEnvironmentMode.VisualStudio);
+                BuildEnvironmentHelper.Instance.MSBuildExtensionsPath.ShouldBe(vsMSBuildDirectory);
+                BuildEnvironmentHelper.Instance.CurrentMSBuildToolsDirectory.ShouldBe(msbuildBinDirectory);
+                BuildEnvironmentHelper.Instance.CurrentMSBuildExePath.ShouldBe(msBuildPath);
+                BuildEnvironmentHelper.Instance.CurrentMSBuildConfigurationFile.ShouldBe(msBuildConfig);
+                BuildEnvironmentHelper.Instance.RunningInVisualStudio.ShouldBeFalse();
```

Comment:
Possibly worth a comment because this confused me. We're not _running_ in VS at this point, but it's a VS instance.